plugins/module_utils/kibana.py

```diff
@@ -70,4 +71,210 @@ def get_alert_connector_by_name(self, connector_name):
+            integration_install = integration_action.install_integration(integration['name'],integration['version'])
+      return(integration_object)
+
+  # Elastic Integration Package Policy functions
+
+  def get_all_pkg_policies(self):
+      endpoint  = 'fleet/package_policies'
+      pkgpolicy_objects = self.send_api_request(endpoint, 'GET')
+      return pkgpolicy_objects
+  
+  def update_pkg_policy(self,pkgpolicy_id,body):
+      endpoint = "fleet/package_policies/" + pkgpolicy_id
+      if not self.module.check_mode:
+        pkg_policy_update = self.send_api_request(endpoint, 'PUT', data=body)
+      else:
+        pkg_policy_update = "Cannot proceed with check_mode set to " + self.module.check_mode
+      return pkg_policy_update
+  
+  def get_pkg_policy(self,agent_policy_id):
+    self.integration_name = self.module.params.get('integration_name')
+    pkg_policy_action = Kibana(self.module)
+    pkg_policy_objects = pkg_policy_action.get_all_pkg_policies()
+    pkg_policy_object = ""
+    for pkgPolicy in pkg_policy_objects['items']:
+      if pkgPolicy['package']['title'] == self.integration_name and pkgPolicy['policy_id'] == agent_policy_id:
+        pkg_policy_object = pkgPolicy
+    return(pkg_policy_object)
+  
+  def create_pkg_policy(self,agent_policy_id, integration_object):
+    self.pkg_policy_name = self.module.params.get('pkg_policy_name')
+    self.pkg_policy_desc = self.module.params.get('pkg_policy_desc')
+    pkg_policy_action = Kibana(self.module)
+    pkg_policy_object = pkg_policy_action.get_pkg_policy(agent_policy_id)
+    if not pkg_policy_object:
+      body = {
+        "name": self.pkg_policy_name,
+        "namespace": "default",
+        "description": self.pkg_policy_desc,
+        "enabled": True,
+        "policy_id": agent_policy_id,
+        "output_id": "",
+        "inputs": [],
+        'package': {
+          'name': integration_object['name'],
+          'title': integration_object['title'],
+          'version': integration_object['version']
+        }
+      }
+      body_JSON = dumps(body)
+      endpoint = 'fleet/package_policies'
+      if not self.module.check_mode:
+        pkg_policy_object = self.send_api_request(endpoint, 'POST', data=body_JSON)
+      else:
+        pkg_policy_object = "Cannot proceed with check_mode set to " + self.module.check_mode
+      return pkg_policy_object
+    else:
+      return pkg_policy_object
+    
+# Elastic Agent Policy functions
+
+  def get_all_agent_policys(self):
+    endpoint  = 'fleet/agent_policies'
+    agent_policy_objects = self.send_api_request(endpoint, 'GET')
+    return(agent_policy_objects)
+
+  def create_agent_policy(self, check_mode=False):
+    agent_policy_action = Kibana(self.module)
+    self.agent_policy_id = self.module.params.get('agent_policy_id')
+    self.agent_policy_name = self.module.params.get('agent_policy_name')
+    self.agent_policy_desc = self.module.params.get('agent_policy_desc')
```

Comment:
Module parameters should be gathered in the module or in `init` if in module_utils as mentioned above.